src/DotVVM.Framework/Resources/Scripts/global-declarations.ts

```diff
@@ -174,3 +175,29 @@ type ValidationRuleTable = {
 type RootViewModel = {
     $csrfToken?: string | KnockoutObservable<string>,
 }
+
+type TypeMap = {
+    [typeId: string]: TypeMetadata
+}
+
+type TypeMetadata = {
+    [prop: string]: PropertyMetadata
+}
+
+type PropertyMetadata = {
+    type: TypeDefinition;
+    post?: "always" | "pathOnly" | "no";
+    update?: "always" | "firstRequest" | "no";
```

Comment:
Is this actually required? We can just trust the server not to send us these properties.